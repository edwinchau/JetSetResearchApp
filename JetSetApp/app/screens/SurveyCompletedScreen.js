import React, { Component } from 'react';
import {Button, Text, View} from 'react-native';

import styles from '../resources/Styles'
import { SurveyAnswersParse, SurveyQuestions } from '../resources/SurveyAnswersParse'
import SaveData from '../resources/SaveData';


export default class SurveyCompletedScreen extends Component {
    static navigationOptions = () => {
        return {
            headerStyle: styles.headerStyle,
            headerTintColor: '#fff',
            // headerTitle: 'Survey Results',
            headerTitleStyle: {
                flex: 1,
            },
            headerLeft: null,
            gesturesEnabled: false
        };
    };

    componentDidMount() {
        this.props.navigation.navigate('Home');
    }

    render() {
        const { navigate } = this.props.navigation;
        let answers = this.props.navigation.getParam('surveyAnswers');
        const surveyQuestions = this.props.navigation.getParam('surveyQuestions');
        console.log(this.props.navigation.getParam('surveyName'));

        let fileSaveLocation = this.props.navigation.getParam('surveyName');

        /**
         * Saves the user data as a JSON instead of a CSV. If there are any files that need to be saved a saved as a JSON
         * include it in the if statement.
         */
        if (this.props.navigation.getParam('surveyName') === "NewUserQuestions") {
            fileSaveLocation = fileSaveLocation + ".json";
            answers = JSON.stringify(answers);

            SaveData.saveNewFile(answers, fileSaveLocation);
        } else {
            fileSaveLocation = fileSaveLocation + ".csv";
            answers = SurveyAnswersParse(answers, surveyQuestions);
            let questions = SurveyQuestions(surveyQuestions);

            SaveData.saveSurveyResults(fileSaveLocation, answers, questions);
        }

        return (
            <View style={styles.background}>
                {/*<View style={styles.container}>*/}
                {/*    <Text>Raw JSON: {JSON.stringify(this.props.navigation.getParam('surveyAnswers', {}))}</Text>*/}
                {/*    <Button*/}
                {/*        onPress={() => navigate('Home')}*/}
                {/*        title="Back Home"*/}
                {/*    />*/}
                {/*</View>*/}
            </View>
        );
    }
}
