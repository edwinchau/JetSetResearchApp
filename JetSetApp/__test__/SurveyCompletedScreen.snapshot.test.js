import React from "react";
import SurveyCompletedScreen from "../app/screens/SurveyCompletedScreen"
import renderer from "react-test-renderer"

describe("<SurveyCompletedScreen/>", ()=>{
  it("SurveyCompletedScreen snapshot", ()=>{
    let wrapper = null;
    const spyNavigate = jest.fn();
    const props = {
      navaigation: {
        navigate: spyNavigate,
        state:{}
      }
    }
    const params = {
      token: 'someToken'
    }

    beforeEach(() => {
      wrapper = renderer.create(<SurveyCompletedScreen {...props} />)
      wrapper.setState({params: params})
    })
    expect(wrapper).toMatchSnapshot();
  });
});
