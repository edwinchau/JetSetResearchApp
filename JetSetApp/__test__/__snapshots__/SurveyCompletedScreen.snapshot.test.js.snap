// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SurveyCompletedScreen/> SurveyCompletedScreen snapshot 1`] = `null`;
